x-pack/plugins/security_solution/public/cases/containers/types.ts

```diff
@@ -137,3 +138,18 @@ export interface FieldMappings {
   id: string;
   title?: string;
 }
+
+export type UpdateKey = keyof Pick<
+  CasePatchRequest,
+  'connector' | 'description' | 'status' | 'tags' | 'title' | 'settings'
+>;
+
+export interface UpdateByKey {
+  updateKey: UpdateKey;
+  updateValue: CasePatchRequest[UpdateKey];
+  fetchCaseUserActions?: (caseId: string) => void;
+  updateCase?: (newCase: Case) => void;
+  caseData: Case;
```

Comment:
We need to know when updating the case if it has alerts. For that reason, we replaced the `version` with the whole `Case` object.